restapi/src/main/java/io/stargate/web/resources/Db.java

```diff
@@ -88,19 +92,28 @@ public AuthenticatedDB getDataStoreForToken(
     AuthenticationSubject authenticationSubject =
         authenticationService.validateToken(token, headers);
     return new AuthenticatedDB(
-        getDataStoreInternal(authenticationSubject, pageSize, pagingState), authenticationSubject);
+        getDataStoreInternal(authenticationSubject, pageSize, pagingState, headers),
+        authenticationSubject);
   }
 
   private DataStore getDataStoreInternal(
-      AuthenticationSubject authenticationSubject, int pageSize, ByteBuffer pagingState) {
+      AuthenticationSubject authenticationSubject,
```

Comment:
There won't be an authenticationSubject for internal. An "internal" datastore is for things like validating a table based token.

Reply:
Never mind, was thinking of `createInternal`.